tests/foreman/ui/test_reporttemplates.py

```diff
@@ -233,6 +234,31 @@ def test_positive_generate_registered_hosts_report(session, module_org, module_l
             assert res['Operating System'] == '{0} {1}'.format(os_name, os.major)
 
 
+@upgrade
+@tier2
+def test_positive_generate_subscriptions_report_json(session, module_org, module_loc):
+    """Use provided Subscriptions report, generate JSON
+
+    :id: b44d4cd8-a88e-47cf-9993-0bb871ac2c96
+
+    :expectedresults: The Subscriptions report is generated in JSON
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Medium
+    """
+    # generate Subscriptions report
+    with session:
+        file_path = session.reporttemplate.generate("Subscriptions",
+                                                    values={'output_format': 'JSON'})
+    with open(file_path) as json_file:
+        data = json.load(json_file)
+        assert len(data) > 0
+        for subscription in data:
+            keys_expected = ['ID', 'Name', 'Available', 'Quantity', 'SKU', 'Contract number']
```

Comment:
Could you move that up, so we don't re-define variable on every loop iteration (I fully expect compiler to optimize it away, but still).